### PR TITLE
Summer2023/add new psb2 instructions

### PR DESCRIPTION
**Pull request by anthonyhevia**

### CODE REVIEW

**Comment by thelmuth** [comment]
Note: this uses the composite types commits, so do this one second and it should merge a lot cleaner.

**Comment by erp12** [comment]
See [this comment](https://github.com/erp12/cbgp-lite/pull/28#pullrequestreview-1520948773) and following thread on #28 for details on how we will merge this.

File: src/erp12/cbgp_lite/lang/lib.clj [hunk 1]
```diff
@@ -3,7 +3,8 @@
   (:require [clojure.core :as core]
             [clojure.set :as set]
             [clojure.string :as str]
-            [erp12.cbgp-lite.lang.schema :as schema]))
+            [erp12.cbgp-lite.lang.schema :as schema]
+            [erp12.cbgp-lite.lang.lib :as lib]))
```
Comment (erp12) [inline src/erp12/cbgp_lite/lang/lib.clj:7]: This namespace probably shouldn't import itself.

File: src/erp12/cbgp_lite/task.clj [hunk 2]
```diff
@@ -16,7 +16,10 @@
 
 (defn vars-for-types
   [types]
-  (set (keys (lib/lib-for-types types))))
+  (let [type-thing 
+        (set (keys (lib/lib-for-types types)))]
+    (println "Vars for this problem:" type-thing)
+    type-thing))
```
Comment (erp12) [inline src/erp12/cbgp_lite/task.clj:22]: I like the idea of logging all the vars used during a run!

I think we should use the logging library rather than a print statement so that we can control if it goes to stdout or a file and we get additional information in the log (timestamp, location in the code, etc). It should be as easy as using `log/info` instead of `println`.

Also, this log statement would probably be more appropriate at the location we start a `run`. Somewhere around [here](https://github.com/erp12/cbgp-lite/blob/master/benchmarks/erp12/cbgp_lite/benchmark/ga.clj#L61-L65), you can get the data to log using `(:vars opts)`. 

The `vars-for-types` function is for getting all variables which only use values of the given `types`. It may not always get called exactly once per run at the start of the run.

File: src/erp12/cbgp_lite/lang/lib.clj
Comment (erp12) [inline src/erp12/cbgp_lite/lang/lib.clj:1]: We should add some tests for the new functions in this [`test/erp12/cbgp_lite/lang/lib_test.clj`](https://github.com/erp12/cbgp-lite/blob/master/test/erp12/cbgp_lite/lang/lib_test.clj). Especially for the edge cases like nan and infinity.

File: src/erp12/cbgp_lite/lang/lib.clj [hunk 3]
```diff
@@ -401,10 +496,20 @@
    ;'min-double          (binary-transform DOUBLE)
    ;'max-int             (binary-transform INT)
    ;'max-double          (binary-transform DOUBLE)
+   'sqrt               (unary-transform DOUBLE)
    `sin                (unary-transform DOUBLE)
    `cos                (unary-transform DOUBLE)
    `tan                (unary-transform DOUBLE)
-   ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+   'asin                (unary-transform DOUBLE)
+   'acos                (unary-transform DOUBLE)
```
Comment (erp12) [inline src/erp12/cbgp_lite/lang/lib.clj:504]: I think these can be \`asin and \`acos  (instead of 'asin and 'acos) and then you won't need an entry in `dealiases`.

File: src/erp12/cbgp_lite/lang/lib.clj [hunk 4]
```diff
@@ -401,10 +496,20 @@
...
+   `atan                (unary-transform DOUBLE)
+   'log2               (unary-transform DOUBLE)
+   'log10              (unary-transform DOUBLE)
```
Comment (erp12) [inline src/erp12/cbgp_lite/lang/lib.clj:507]: Same here. \`log2 and \`log10 should work with no dealias.

**Review by erp12 — CHANGES_REQUESTED** [review]

File: src/erp12/cbgp_lite/task.clj (hunk 2 above)
Comment (thelmuth) [reply src/erp12/cbgp_lite/task.clj:22]: Does `(:vars opts)` contain only variables which use values of the given `types`? If so, good, cause that's why we were printing it in `vars-for-types`
